libtransmission/session.cc

```diff
@@ -2244,6 +2252,61 @@ int tr_sessionGetCacheLimit_MB(tr_session const* session)
 ****
 ***/
 
+void tr_session::setDefaultTrackers(std::string_view trackers)
```

Comment:
I had automated tracker tiers, a few weeks ago, until `tor->announceList().add()` lost its ability to pass a "tier" parameter.

That parser of yours is way too conservative for random URL lists on the Internet.

Reply:
> That parser of yours is way too conservative for random URL lists on the Internet.

It's currently being used to parse utorrent-style tracker lists, so if there are URLs that it can't parsed, that needs to be fixed. Do you have examples of lists that break it?

Reply:
Can't find any. Just trying to follow https://en.wikipedia.org/wiki/Robustness_principle

Reply:
If you can't find anything that breaks it, how is it too conservative for use here?

Reply:
I don't really care how it's done, as long as I stop wasting time maintaining this patch, while getting to keep using the functionality.

So feel free to improve it after merging, any way you see fit. I only want default trackers for public torrents in Transmission and a nice UI to manage them. The rest are details.

Reply:
Well, that answers the question of intent.

Reply:
> I had automated tracker tiers, a few weeks ago, until `tor->announceList().add()` lost its ability to pass a "tier" parameter.

It didn't though? https://github.com/transmission/transmission/blob/02b6cc76d197ae74aed60f2f824385e9cfa619f4/libtransmission/announce-list.h#L100

Reply:
Oh, you just switched param order.

Then why did the tier disappear from what I was modelling from?

https://github.com/transmission/transmission/commit/52f6e80a61ce7340c6e9a60689e345975f02bf8d#diff-6b3564ae830bc5febeda0f04be5f5043ad7a0880df7f034d5538db0a2144ea93L1056

I see that `tr_announce_list::nextTier()` is still there, so we can bring the one-tier-per-tracker-URL scheme back, if you want it.

Reply:
> Then why did the tier disappear from what I was modelling from?

When a tier isn't specified, `nextTier()` is used as the default. That's why it has to go last, since the argument has a default value

Reply:
> I see that `tr_announce_list::nextTier()` is still there, so we can bring the one-tier-per-tracker-URL scheme back, if you want it.

Changing it wouldn't be useful now. When you wrote afternoon that I should do the rest of the details myself, I made a new branch using https://github.com/transmission/transmission/pull/229.patch as a starting point & keeping your authorship in the commit history.

I don't know how you feel about this patch or the project's maintainers after so long, but FWIW I appreciate the work you've done on this.  **Very few** PR submitters take the time to make a new feature work on multiple clients. And add updating it to C++ / gtkmm as well. You did a good job and I appreciate it.